ports:
    thruster:"/dev/ttyUSB0"
thrusters:
  -   name: "left"
      address: 128
      port: 0
      orientation:
          x: 0.0
          y: 0.0
          z: 1.0
      position:
          x: 0.0
          y: 0.25
          z: 0.0

  -   name: "right"
      address: 128
      port: 1
      orientation:
          x: 0.0
          y: 0.0
          z: -1.0
      position:
          x: 0.0
          y: -0.25
          z: 0.0
  -   name: "top"
      address: 129
      port: 0
      orientation:
          x: -1.0
          y: 0.0
          z: 0.0
      position:
          x: 0.0
          y: 0.0
          z: 0.25
  -   name: "bottom"
      address: 129
      port: 1
      orientation:
          x: 1.0
          y: 0.0
          z: 0.0
      position:
          x: 0.0
          y: 0.0
          z: -0.25
  -   name: "front"
      address: 130
      port: 0
      orientation:
          x: 0.0
          y: 1.0
          z: 0.0
      position:
          x: 0.25
          y: 0.0
          z: 0.0
  -   name: "back"
      address: 130
      port: 1
      orientation:
          x: 0.0
          y: -1.0
          z: 0.0
      position:
          x: -0.25
          y: 0.0
          z: 0.0
control:
    rate: 10
    mass: 25
    back_thrust_ratio: 1.0
    max_thrust: 25
    inertia:
        psi: 1.0
        phi: 1.0
        theta: 1.0
    proportional:
        x: 1.0
        y: 1.0
        z: 20.0
        psi: 30.0 #roll
        phi: 50.0 #pitch
        theta: 20.0 #yaw
    integral:
        x: 0
        y: 0
        z: 0
        psi: 0
        phi: 0
        theta: 0
    derivative:
        x: 0
        y: 0
        z: 0
        psi: 0
        phi: 0
        theta: 0
    windup:
        x: 0
        y: 0
        z: 0
        psi: 0
        phi: 0
        theta: 0
    hysteresis:
        x: 0
        y: 0
        z: 0
        psi: 0
        phi: 0
        theta: 0
    limits:
        rotation: 0.6
        translation: 0.8
